### PR TITLE
Ft signin fullstack 157912677

### PR DESCRIPTION
**Pull request by LuffyKing**
#### What does this PR do?
It sets up the client branch and implements login
#### Description of Task to be completed?
Creation of a client folder for the full stack version of TrackerHero. Creation of a login feature for the application that integrates the front end login page and the login API, this will allow users to log in to TrackerHero. Creation of a route and fetch request to protect pages that require a logged in user.
#### What are the relevant Github Issues?
The only issue so is only the pending review
Questions:

### CODE REVIEW

**Comment by benfluleck** [comment]
Why are the 2 commits having the same title? Please fix

File: client/UI/css/AdminViewRequests.css [hunk 1]
```diff
@@ -0,0 +1,78 @@
+.modalDelResolve,.approvebtn, .modalDeldisapprove,.modalCancel,
+.modalDelApprove,.disapprovebtn, .resolvebtn,.placeholder,
```
Comment (benfluleck) [inline client/UI/css/AdminViewRequests.css]: This is not proper css naming convention, please fix

File: client/UI/html/AdminUserRequestDetailPage.html [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  <link rel="stylesheet" href="../css/AdminViewRequests.css">
+  <link rel="stylesheet" href="../css/UserRequestDetail.css">
+</head>
+<body>
+  <header>
+    <nav id="navbar">
+      <a class="default-primary-color text-primary-color Logo pacifico" href="SigninPage.html">TrackerHero</a>
+      <ul class="default-primary-color text-primary-color">
+        <li><a href="MakeRequest.html">Make Request &nbsp<i class="fas fa-bullhorn"></i></a></li>
+        <li><a href="#">Profile &nbsp<i class="fas fa-user"></i></a></li>
```
Comment (benfluleck) [inline client/UI/html/AdminUserRequestDetailPage.html]: please use padding and not nbsp

File: client/UI/html/SigninPage.html [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  <link rel="stylesheet" href="../css/LandingPage.css">
+  <link rel="stylesheet" href="../css/signin.css">
+</head>
+<body class="landing" id="overlay">
+  <header>
+    <nav id="navbar">
+      <a class="default-primary-color text-primary-color Logo pacifico" href="SigninPage.html">TrackerHero</a>
+      <ul class="default-primary-color text-primary-color">
+        <li><a href="MakeRequest.html">Make Request &nbsp<i class="fas fa-bullhorn"></i></a></li>
+        <li><a href="#">Profile &nbsp<i class="fas fa-user"></i></a></li>
```
Comment (benfluleck) [inline client/UI/html/SigninPage.html]: Please change all instances where you use nbsp

File: client/UI/js/alertAdminViewRequests.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    headers: new Headers({
+      authorization: localStorage.getItem('token'),
+     'Content-Type': 'application/json'
+   }),
+    body: JSON.stringify({
+      email: form.elements.email.value,
+      password: form.elements.password.value
+    })
+  })
+    .then(response => response.json()).then((json) => {
```
Comment (benfluleck) [inline client/UI/js/alertAdminViewRequests.js]: Maybe a different name than json would be more appropriate here

File: client/server/validation/DatabaseHelpers.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    if (error) {
+      response.status(500).send({ message: error.stack });
+    }
+    client.query(query, value, (error1, requestRow) => {
+      done();
+      if (error1) {
+        return response.status(500).send({ message: error1.stack });
+      }
+      if (requestRow.rows.length > 0) {
+        const requestPluSing = requestRow.rows.length === 1 ? 'request has' : 'requests have';
```
Comment (benfluleck) [inline client/server/validation/DatabaseHelpers.js]: what is requestPluSing?

File: client/server/validation/DatabaseHelpers.js [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+        if (operation === 'get multiple requests') {
```
Comment (benfluleck) [inline client/server/validation/DatabaseHelpers.js]: in this case statement would implement this a lot quicker but also why are we tightly coupling the operation here using a numeric value with a case statement would be the ideal scenario here.

File: client/server/validation/RejectStatusValidator.js [hunk 7]
```diff
@@ -0,0 +1,13 @@
+import { statusChanger } from './DatabaseHelpers';
+/**
+* It sets the status to Rejected
+* @param {Object} request - request object containing params and body
```
Comment (benfluleck) [inline client/server/validation/RejectStatusValidator.js]: why are the param in capital go and check the difference between object and  Object in JS

File: client/server/validation/createARequestValidator.js [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+      reqBody[key] = reqBody[key].toLowerCase().replace(/^\w/, firstLetter => firstLetter.toUpperCase());
+    }
+    reqBody[key] = reqBody[key].trim();
+  });
+  request.reqBody = reqBody;
+};
+/**
+* It finds the fields that are undefined or null
+* @param {Object} reqBody - object containing the relevant field values
+* @returns {string[]} - an array of the undefined fields as strings
```
Comment (benfluleck) [inline client/server/validation/createARequestValidator.js]: I am pretty sure this is not a type in javascript, please adjust

File: client/server/validation/createARequestValidator.js [hunk 9]
```diff
@@ -0,0 +1,174 @@
...
+const specialMessages = {
+  userid: 'the userid value is not UUID',
+  type: 'the type value is not Repair or Maintenance',
+  email: 'the email value is not an email'
+};
+/**
+* It finds the fields that are supposed to have a string as a value but
+* possesses any data type instead
+* @param {number} statusCode - status code
+* @param {Object} response - response object that conveys the result of the request
```
Comment (benfluleck) [inline client/server/validation/createARequestValidator.js]: Sometimes small letter object sometime Object, which one you dey?

File: client/server/validation/createARequestValidator.js [hunk 10]
```diff
@@ -0,0 +1,174 @@
...
+* @param {string[]} invalidFieldsArray - an array of invalid fields
```
Comment (benfluleck) [inline client/server/validation/createARequestValidator.js]: see above

File: client/server/validation/createARequestValidator.js [hunk 11]
```diff
@@ -0,0 +1,174 @@
...
+  const emptyFieldsArr = emptyFieldsFinder(reqBody);
+  if (emptyFieldsArr.length > 0) {
+    return message(400, response, emptyFieldsArr, 'not provided', failReason);
+  }
+};
+/**
+* It validates the fields and renders the appropriate response
+* @param {Object} request - request object containing params and body
+* @param {Object} response - response object that conveys the result of the request
+* @param{Object} next - middleware that calls the net middleware in the stack
```
Comment (benfluleck) [inline client/server/validation/createARequestValidator.js]: What is going on here please add the space

File: client/server/validation/DatabaseHelpers.js (hunk 5 above)
Comment (LuffyKing) [reply client/server/validation/DatabaseHelpers.js]: The text changes based on the number of requests, If it it plural it becomes 'requests have'.

File: client/server/validation/createARequestValidator.js (hunk 8 above)
Comment (LuffyKing) [reply client/server/validation/createARequestValidator.js]: That says the input is a string of arrays

File: client/server/validation/DatabaseHelpers.js (hunk 5 above)
Comment (benfluleck) [reply client/server/validation/DatabaseHelpers.js]: and you decided the best name for such a variable was a name nobody could understand. Please refactor

File: client/server/validation/createARequestValidator.js (hunk 8 above)
Comment (benfluleck) [reply client/server/validation/createARequestValidator.js]: So this is actually not a type in javascript but it is in typescript. Take note also refer to types in javascript.

File: client/server/validation/createARequestValidator.js (hunk 11 above)
Comment (LuffyKing) [reply client/server/validation/createARequestValidator.js]: lol

**Pull request merged**